rectools/visuals/metrics_app.py

```diff
@@ -149,61 +258,106 @@ def _update_chart(
         metric_y: widgets.Dropdown,
         use_avg: widgets.Checkbox,
         fold_i: widgets.Dropdown,
-    ) -> None:
-        data = self._make_chart_data_avg() if use_avg.value else self._make_chart_data(fold_i.value)
-        existing_traces = {trace.name: trace for trace in fig_widget.data}
+        meta_feature: widgets.Dropdown,
+        use_meta: widgets.Checkbox,
+    ) -> None:  # pragma: no cover
+        chart_data = self._create_chart_data(use_avg, fold_i)
+        color_clmn = meta_feature.value if use_meta.value else Columns.Model
+
+        # Save dots symbols from the previous widget state
+        # `split(" ", 1)[-1]` removed metainfo from trace name. Thus we guarantee to map with traces from previous state
+        trace_name2symbol = {trace.name.split(" ", 1)[-1]: trace.marker.symbol for trace in self.fig.data}
```

Comment:
About the splitting:
1. Trace name starts from param name. And we didn't replace `" "` to `"_"` for parameters. So we might have a problem
2. We don't need to replace `" "` to `"_"` at all (I think). We need to replace `","` to `"_"` or something else. Or replace `", "` to `",_"`. And after that we need to split by `", "` which will solve our problem
3. Also please move this to a separate func so that you write this logic only one time, not three time in the code

Reply:
1. We replaced `"  "` to `"_"` in line [135](https://github.com/MobileTeleSystems/RecTools/blob/38ef219326826ffd64be8acba54c124cceec973a/rectools/visuals/metrics_app.py#L135). 
2. In this case we have to check if user activate "color by meta" mode or not. Otherwise we could trim model name.
3. Sure, good point

Reply:
1. We replaced them for model names but not for meta param names which are first in trace.name when "color by meta" is on. How about splitting `trace.name.split(", ")[-1]`? This way we don't care what's in meta param names, only model names need to be processed
2.I think it's not obligatory if we process and split correctly